spec/setup/helpers/abs_helper_spec.rb

```diff
@@ -0,0 +1,418 @@
+
+        expect(subject.abs_is_valid_response?(test_http_response)).to eq(false)
+      end
+
+    end
+
+
+  end
+
+  describe '#abs_reformat_resource_host' do
+
+    context 'when a valid response_body is provided' do
+
+      it 'returns the expected reformatted response' do
+        expect(subject.abs_reformat_resource_host(TEST_AWSDIRECT_RESPONSE_BODY)).to eq(TEST_REFORMATTED_RESPONSE_BODY)
+      end
+
+    end
+
+    context 'when an invalid response_body is provided' do
+
+      it 'reports the error and returns nil' do
+        expect(subject).to receive(:puts).with('JSON::ParserError encountered')
+        expect(subject.abs_reformat_resource_host(TEST_INVALID_RESPONSE_BODY)).to eq(nil)
+
+      end
+
+    end
+
+  end
+
+  describe '#abs_get_resource_hosts' do
+
+    context 'when a valid response is returned' do
+
+      it 'sets the environment variable, logs the hosts, and returns the response(s)' do
+        ENV['ABS_TOKEN'] = TEST_ABS_TOKEN
+        expect(subject).to receive(:abs_request_awsdirect).at_least(:once).and_return(test_http_response)
+
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_VALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_AWSDIRECT_RESPONSE_BODY)
+
+        expect(subject.abs_get_resource_hosts).to eq(TEST_ABS_RESOURCE_HOSTS)
+      end
+
+    end
+
+    context 'when an invalid response code is returned' do
+
+      it 'reports the error and returns nil' do
+        ENV['ABS_TOKEN'] = TEST_ABS_TOKEN
+        test_expected_result = nil
+
+        expect(subject).to receive(:abs_request_awsdirect).at_least(:once).and_return(test_http_response)
+
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_INVALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_INVALID_RESPONSE_BODY)
+
+        expect(subject.abs_get_resource_hosts).to eq(test_expected_result)
+      end
+
+    end
+
+    context 'when an invalid response body is returned' do
+
+      it 'reports the error and returns nil' do
+        ENV['ABS_TOKEN'] = TEST_ABS_TOKEN
+        test_expected_result = nil
+
+        expect(subject).to receive(:abs_request_awsdirect).at_least(:once).and_return(test_http_response)
+
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_VALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_INVALID_RESPONSE_BODY)
+
+        expect(subject.abs_get_resource_hosts).to eq(test_expected_result)
+      end
+
+    end
+
+    context 'when the response is nil' do
```

Comment:
This test can be dropped if the above mocks abs_is_valid_response. This method doesn't differentiate between bad code and bad body, and nil.

Reply:
This should be resolved. The method now reports the code and body when the response is not nill, and the spec tests each scenario.

Reply:
I think this got fixed, and somehow the comment remained, as it doesn't seem to apply to the code that it is attached to anymore... (mocking abs_is_valid_response when testing it  wouldn't make any sense... )